code/Test_definitions/checkDeviceSwap.feature

```diff
@@ -0,0 +1,213 @@
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has been swapped in the last 240 hours
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == true
+
+    @check_device_swap_3_valid_device_swap_max_age
+    Scenario Outline: Check that the response shows that the device has been swapped - maxAge is provided in the request
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has been swapped in the last "<hours>"
+        And the request body property "maxAge" is set to a value equal or greater than "<hours>" within the allowed range
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == true
+
+        Examples:
+            | hours |
+            | 260   |
+            | 120   |
+            | 24    |
+            | 12    |
+
+    @check_device_swap_4_more_than_240_hours
+    Scenario: Check that the response shows that the device has not been swapped when the last swap was more than 240 hours ago
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has been swapped more than 240 hours ago
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_5_out_of_max_age
+    Scenario: Check that the response shows that the device has not been swapped when the last swap was before the maxAge field
+        Given a valid phone number identified by the token or provided in the request body
+        And the last swap for this phone number's in the device is known
+        And the request body property "maxAge" is set to a value lower that the last known device swap
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_6_no_device_swap_no_max_age
+    Scenario: Check that the response shows that the device has not been swapped - maxAge is not provided in the request parameter
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has never been swapped
+        When the request "checkDeviceSwap" is sent
+        Then the response status code is 200
+        And the value of response property "$.swapped" == false
+
+    @check_device_swap_7_no_device_swap_with_max_age
+    Scenario Outline: Check that the response shows that the device has never been swapped - maxAge is provided in the request
+        Given a valid phone number identified by the token or provided in the request body
+        And the device has never been swapped
```

Comment:
Idem previous - should we add that the link sim-device is also anterior?

Reply:
included